go-controller/pkg/factory/factory.go

```diff
@@ -309,6 +317,31 @@ type WatchFactory struct {
 
 	iFactory  informerfactory.SharedInformerFactory
 	informers map[reflect.Type]*informer
+	ObjectCacheInterface
+}
+
+// ObjectCacheInterface represents the exported methods for getting
+// kubernetes resources from the informer cache
+
+type ObjectCacheInterface interface {
+	GetPod(namespace, name string) (*kapi.Pod, error)
+	GetPods(namespace string) ([]*kapi.Pod, error)
+	GetNodes() ([]*kapi.Node, error)
+	GetNode(name string) (*kapi.Node, error)
+	GetService(namespace, name string) (*kapi.Service, error)
+	GetEndpoints(namespace string) ([]*kapi.Endpoints, error)
+	GetEndpoint(namespace, name string) (*kapi.Endpoints, error)
+	GetNamespaces() ([]*kapi.Namespace, error)
+}
+
+// objectCache is the structure upon which the
+// ObjectCacheInterface is implemented
+type objectCache struct {
```

Comment:
@abhat let's just put the listers into the informer itself rather than having a separate struct for them all?

Reply:
I had thought about it. But then saw the `informer` object grabs locks for all the public API functions. I wanted to steer clear of having to use the lister object without following the proper locking semantics.

Reply:
@abhat we don't need to lock when using the lister though. We only need to lock when using some of the other internal fields.